src/libcore/char.rs

```diff
@@ -448,19 +459,15 @@ impl Iterator for EscapeUnicode {
                 Some('u')
             }
             EscapeUnicodeState::LeftBrace => {
-                let mut n = 0;
-                while (self.c as u32) >> (4 * (n + 1)) != 0 {
-                    n += 1;
-                }
-                self.state = EscapeUnicodeState::Value(n);
+                self.state = EscapeUnicodeState::Value;
                 Some('{')
             }
-            EscapeUnicodeState::Value(offset) => {
-                let c = from_digit(((self.c as u32) >> (offset * 4)) & 0xf, 16).unwrap();
-                if offset == 0 {
+            EscapeUnicodeState::Value => {
+                let c = from_digit(((self.c as u32) >> (self.offset * 4)) & 0xf, 16).unwrap();
```

Comment:
Ah, it also requires storing `c` as a `u32`, as it might contain values that are illegal for a `char`. Experimenting now with this. Thank you for the suggestions :)